trie/iterator.go

```diff
@@ -31,19 +32,25 @@ type Iterator struct {
 // NewIterator creates a new key-value iterator.
 func NewIterator(trie *Trie) *Iterator {
 	return &Iterator{
-		trie:   trie,
 		nodeIt: NewNodeIterator(trie),
-		keyBuf: make([]byte, 0, 64),
+		Key:    nil,
+	}
+}
+
+// FromNodeIterator creates a new key-value iterator from a node iterator
+func FromNodeIterator(it NodeIterator) *Iterator {
```

Comment:
I think the correct name is `NewIteratorFromNodeIterator`. A bit unwieldy but `NewSomething` should be part of the method signature to make it visible what it does. I'm open to suggestions though :D